server/src/main/java/org/opensearch/node/resource/tracker/AverageIoUsageTracker.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.node.resource.tracker;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.Constants;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.monitor.fs.FsInfo.DeviceStats;
+import org.opensearch.monitor.fs.FsService;
+import org.opensearch.node.IoUsageStats;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.HashMap;
+
+/**
+ * AverageIoUsageTracker tracks the IO usage by polling the FS Stats for IO metrics every (pollingInterval)
+ * and keeping track of the rolling average over a defined time window (windowDuration).
+ */
+public class AverageIoUsageTracker extends AbstractAverageUsageTracker {
+
+    private static final Logger LOGGER = LogManager.getLogger(AverageIoUsageTracker.class);
+    private final FsService fsService;
+    private final HashMap<String, Long> prevIoTimeDeviceMap;
+    private long prevTimeInMillis;
+    private IoUsageStats ioUsageStats;
+
+    public AverageIoUsageTracker(FsService fsService, ThreadPool threadPool, TimeValue pollingInterval, TimeValue windowDuration) {
+        super(threadPool, pollingInterval, windowDuration);
+        this.fsService = fsService;
+        this.prevIoTimeDeviceMap = new HashMap<>();
+        this.prevTimeInMillis = -1;
+        this.ioUsageStats = null;
+    }
+
+    /**
+     * Get current IO usage percentage calculated using fs stats
+     */
+    @Override
+    public long getUsage() {
+        long usage = 0;
+        if (this.preValidateFsStats()) {
+            return usage;
+        }
+        // Currently even during the raid setup we have only one mount device and it is giving 0 io time from /proc/diskstats
+        DeviceStats[] devicesStats = fsService.stats().getIoStats().getDevicesStats();
+        long latestTimeInMillis = fsService.stats().getTimestamp();
```

Comment:
We should use nano time as logic time rather than wall clock time

Reply:
Agree. As discussed we have check for difference between previous time and current time should be greater than 0.  Incase if the clock time becomes negative we won't consider that.